### ISSUE TITLE
GaussJacobi panics for some valid inputs

### ISSUE THREAD

**Issue by JSorngard** [DomiDre/gauss-quad#51]
`GaussJacobi::new(10, -0.5, -0.5)` panics because the nodes and weights are evaluated to `NaN`, and then the comparison panics in the call to `sort_unstable_by`. I have added a test that finds this bug in the branch `found_nans`.

**Comment by JSorngard** [comment]
`GaussJacobi::new(anything, -1.0, -1.0)` also panics due to line 72: `(2.0 + alpha + beta)`. When alpha = beta = -1 this is a division by zero.

**Comment by JSorngard** [comment]
The division by zero is fixed in #55.

**Comment by DomiDre** [comment]
Good catch. Looking again at the formula, alpha, beta > -1 is correct. The =  must have slipped in some way

**Comment by JSorngard** [comment]
In the case of `GaussJacobi::new(deg, -0.5, -0.5)` it panics because the first time through the loop `denom_sum` is 1 and the off diagonal element has a denominator of `(denom_sum + 1.0) * (denom_sum - 1.0)`, resulting in a division by zero.

**Comment by JSorngard** [comment]
I think unfortunately this is still an issue, but now only when alpha=beta!=-1
